purchases/src/test/java/com/revenuecat/purchases/PurchasesTest.kt

```diff
@@ -1634,337 +1609,50 @@ class PurchasesTest: BasePurchasesTest() {
     // region offerings
 
     @Test
-    fun `fetch offerings on foregrounded if it's stale`() {
```

Comment:
So much better removing code from here 👍🏻 
This file is still 4100 lines. We could do the same I did in iOS:
- https://github.com/RevenueCat/purchases-ios/pull/1609
- https://github.com/RevenueCat/purchases-ios/pull/1610
- https://github.com/RevenueCat/purchases-ios/pull/1611
- https://github.com/RevenueCat/purchases-ios/pull/1612
- https://github.com/RevenueCat/purchases-ios/pull/1613
- https://github.com/RevenueCat/purchases-ios/pull/1628
- https://github.com/RevenueCat/purchases-ios/pull/1629
- https://github.com/RevenueCat/purchases-ios/pull/1630
- https://github.com/RevenueCat/purchases-ios/pull/1632
- https://github.com/RevenueCat/purchases-ios/pull/1633
- https://github.com/RevenueCat/purchases-ios/pull/1637
- https://github.com/RevenueCat/purchases-ios/pull/1638
- https://github.com/RevenueCat/purchases-ios/pull/1639

Reply:
A lot of that code in iOS is no longer in `Purchases`, but we have a lot of tests since this is the facade so it's useful to test from top to bottom.

Reply:
Yeah, we definitely need to improve these tests situation. We already have a first step with #1011 (Thanks @vegaro!).

As for keeping tests at the `Purchases` level, I think that makes sense to test the main flows and main things we should be careful of. I guess that would be pretty close to integration tests, since at that point we probably only want to mock shared preferences/backend... I think that would be a separate test suite since they might be slightly harder to write, but we can discuss it further.

Also, thanks for sharing those! It can definitely serve as inspiration